### ISSUE TITLE
Add first-class one-class / novelty-detection estimator (fit on a single class, score by anomaly membership)

### ISSUE THREAD

**Issue by fundthmcalculus** [fundthmcalculus/tribble-fis#104]
## Motivation

The "none of the above" anomaly mechanism is one of TribbleFIS's most distinctive
features, but today it is only reachable *through a multi-class classifier*. There
is no supported way to fit on a **single class** ("normal") and get an anomaly /
novelty score for new points — the classic one-class / outlier-detection setup.

Concrete use case that hit this: monitoring a small LM's per-layer activations to
flag anomalous / adversarial prompts. You have abundant benign traffic and want a
detector trained on benign only that scores how far a new prompt sits from
"normal activity." This is exactly what the fuzzy anomaly rule *should* be ideal
for, but the library cannot express it directly, so it had to be reimplemented by
hand — which then diverges from the real mechanism (different feature selection,
single Gaussians vs mixtures, different norm aggregation).

## Current behaviour (v0.1.0)

`TribbleClassifier` accepts a single-class `y` without error, but the resulting
model is unusable for anomaly detection:

```python
import numpy as np, pandas as pd
from tribblefis.gaussian_classifier import TribbleClassifier

rng = np.random.default_rng(0)
X = pd.DataFrame(rng.normal(0, 1, size=(200, 6)), columns=[f"f{i}" for i in range(6)])
y = np.zeros(200, dtype=int)                 # ONE class: "normal"
clf = TribbleClassifier(top_n=4, n_gaussians=2).fit(X, y)

X_normal  = pd.DataFrame(rng.normal(0, 1, size=(50, 6)), columns=X.columns)
X_anom    = pd.DataFrame(rng.normal(6, 1, size=(50, 6)), columns=X.columns)  # far OOD

clf.predict_proba(X_normal)[:, 0].mean()   # -> 1.0
clf.predict_proba(X_anom)[:,   0].mean()   # -> 1.0   (identical! no signal)
```

`predict_proba` returns `1.0` for everything, normal and wildly out-of-distribution
alike, because the softmax over a single class is trivially 1. So the public API
gives no anomaly signal.

**The signal does exist internally**, though — the raw firing strength already
separates the two:

```python
fn, _ = clf.firing_strengths(X_normal)
fa, _ = clf.firing_strengths(X_anom)
np.max(fn, 1).mean()   # 0.275  (normal points fire the learned rules)
np.max(fa, 1).mean()   # 0.000  (far points fire nothing)
```

So `1 - max_rule(firing_strength)` is already a perfectly good novelty score. The
gap is that nothing wraps it as a supported estimator.

## Two things are missing

1. **A one-class estimator / scoring API.** An `OutlierMixin`-style class (or a
   mode on the existing one) that fits on a single class and exposes:
   - `score_samples(X)` / `decision_function(X)` returning the anomaly score
     (`1 - max firing`, or the `AnomalyParameters` complement `1 - max membership - theta`),
   - `predict(X)` returning inlier/outlier under a threshold,
   so it composes with sklearn (`roc_auc_score`, pipelines, `OutlierMixin`).

2. **Feature selection that works with one class.** The current
   differentiation-strength ranking scores features by *class separation*, which
   is undefined with a single class (it silently ranks near-constant features
   first — in the activation use case it put an almost-constant feature at score
   1.0). One-class fitting needs an unsupervised alternative: e.g. rank by
   per-feature dispersion / tightness of the normal cluster, accept a
   user-supplied feature list, or fall back to "use all features" with a warning.

## Suggested shape

```python
from tribblefis.anomaly import TribbleOneClassDetector   # OutlierMixin

det = TribbleOneClassDetector(
    n_gaussians=2,
    norm_conorm="probability",
    feature_selection="dispersion",   # unsupervised; or "all", or explicit list
    contamination=0.05,               # sets the predict() threshold, sklearn-style
)
det.fit(X_normal)                     # single class, no labels
scores = det.score_samples(X_test)    # higher = more anomalous
labels = det.predict(X_test)          # +1 inlier / -1 outlier
```

Internally this can reuse almost everything already there — the Gaussian
membership fitting, `firing_strengths`, and `AnomalyParameters` — so it is mostly
a thin one-class wrapper plus the unsupervised feature-selection path.

## Why it is worth doing

- It makes the fuzzy anomaly rule usable in the setting it is most naturally
  suited to (novelty detection with abundant normal data, few or no anomalies),
  and lets it be benchmarked head-to-head against Mahalanobis / isolation forest /
  one-class SVM without a hand-rolled reimplementation.
- The `1 - max_firing` mechanism already works; this is packaging + an
  unsupervised feature-selection path, not new theory.
- It gives the interpretable-rule story a foothold in outlier detection: "this
  point is anomalous because it fires no learned normal-pattern rule."

Happy to help with a PR if the API direction is agreeable.
